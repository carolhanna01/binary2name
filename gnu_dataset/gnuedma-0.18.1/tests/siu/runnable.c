/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <edma.h>

int test();

main(int argc,char *argv[])
{
  ESint32          i;
  OBJID            id;
  OBJID            id_sc;

  /* Initialize EDMA System*/
  EDMAInit();

  /* Put your C code here */
  edma_printf ("\n Runnable SIU Proxy Test");
  edma_printf ("+ Creating SUPER_CLASS object...");
  if ((id = edma_new_obj ("RUNABLE:SUPER_CLASS")) == -1)  
    {
      edma_printf ("Can't create SUPER_CLASS object");
      EDMAEnd();
      exit(1);
    }
  edma_printf ("Object Report for SUPER_CLASS object");
  edma_obj_report (id);
  edma_printf ("--------------------------------");
  edma_rprop3 (id, "id", &id_sc);
  edma_over_met (id_sc, "Met1", NULL, (PPROC)test);
  edma_met3 (id, "Met1");
  for (i = 0; i < 20; sleep(1), i++)
    edma_printf ("[INFO-main] Iteration %d on main thread",i);

  edma_free_obj (id);

  /* Shutdown EDMA Suystem */
  EDMAEnd();
  return 0;
}

/* Test method for runnable SIU Proxy*/
int test() {
  int     i;
  
  for (i=0;i<10;sleep(1),i++)
    edma_printf ("[INFO-test] Iteration %d on other thread",i);

  return 0;
}
