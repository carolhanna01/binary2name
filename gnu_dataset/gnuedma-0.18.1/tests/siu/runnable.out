
GNU/EDMA 0.17.1 build [Feb  6 2010][Beta][0.17.1] starting up...
EDMA Startup process... OK

 Runnable SIU Proxy Test
+ Creating SUPER_CLASS object...
(RUNABLE)Creating object of class SUPER_CLASS
Object Report for SUPER_CLASS object
==OBJECT REPORT==================================================
Object (0) of class (RUNABLE)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
--------------------------------
(RUNABLE)Creating thread for method Met1
[RUNABLE] Invoking threaded method Met1 on object 1
[INFO-test] Iteration 0 on other thread
[INFO-main] Iteration 0 on main thread
[INFO-test] Iteration 1 on other thread
[INFO-main] Iteration 1 on main thread
[INFO-test] Iteration 2 on other thread
[INFO-main] Iteration 2 on main thread
[INFO-test] Iteration 3 on other thread
[INFO-main] Iteration 3 on main thread
[INFO-test] Iteration 4 on other thread
[INFO-main] Iteration 4 on main thread
[INFO-test] Iteration 5 on other thread
[INFO-main] Iteration 5 on main thread
[INFO-test] Iteration 6 on other thread
[INFO-main] Iteration 6 on main thread
[INFO-test] Iteration 7 on other thread
[INFO-main] Iteration 7 on main thread
[INFO-test] Iteration 8 on other thread
[INFO-main] Iteration 8 on main thread
[INFO-test] Iteration 9 on other thread
[INFO-main] Iteration 9 on main thread
[edma_thread_unregister] Unregistering thread 0xb7d11b90
======> Index 1 of 2[edma_thread_unregister] Thread 0xb7d11b90 found at index 1
[INFO-main] Iteration 10 on main thread
[INFO-main] Iteration 11 on main thread
[INFO-main] Iteration 12 on main thread
[INFO-main] Iteration 13 on main thread
[INFO-main] Iteration 14 on main thread
[INFO-main] Iteration 15 on main thread
[INFO-main] Iteration 16 on main thread
[INFO-main] Iteration 17 on main thread
[INFO-main] Iteration 18 on main thread
[INFO-main] Iteration 19 on main thread
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
